hook-common/src/kafka_messages/app_metrics.rs

```diff
@@ -77,19 +88,88 @@ where
     serializer.serialize_str(category_str)
 }
 
-fn serialize_error_type<S>(error_type: &ErrorType, serializer: S) -> Result<S::Ok, S::Error>
+fn deserialize_category<'de, D>(deserializer: D) -> Result<AppMetricCategory, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: String = Deserialize::deserialize(deserializer)?;
+
+    let category = match &s[..] {
+        "processEvent" => AppMetricCategory::ProcessEvent,
+        "onEvent" => AppMetricCategory::OnEvent,
+        "scheduledTask" => AppMetricCategory::ScheduledTask,
+        "webhook" => AppMetricCategory::Webhook,
+        "composeWebhook" => AppMetricCategory::ComposeWebhook,
+        _ => {
+            return Err(serde::de::Error::unknown_variant(
+                &s,
+                &[
+                    "processEvent",
+                    "onEvent",
+                    "scheduledTask",
+                    "webhook",
+                    "composeWebhook",
+                ],
+            ))
+        }
+    };
+
+    Ok(category)
+}
+
+fn serialize_error_type<S>(error_type: &Option<ErrorType>, serializer: S) -> Result<S::Ok, S::Error>
```

Comment:
nit: I may be wrong but I think there is a Serialize/Deserialize implementation for `Option` and `&Option`, so we could implement Serialize/Deserialize on `ErrorType` instead and we would get Serialize/Deserialize on `&Option<ErrorType>` too. Not a big deal, but would save the check for `Some(error_type)`/`None`

Reply:
Hmmm, maybe I need a special serde annotation to make it work?

```
error[E0308]: mismatched types
   --> hook-common/src/kafka_messages/app_metrics.rs:46:23
    |
46  | #[derive(Deserialize, Serialize, Debug, PartialEq, Clone)]
    |                       ^^^^^^^^^ expected `ErrorType`, found `&Option<ErrorType>`
...
67  |         serialize_with = "serialize_error_type",
    |                          ---------------------- arguments to this function are incorrect
    |
    = note:   expected enum `ErrorType`
            found reference `&'__a std::option::Option<ErrorType>`
note: function defined here
   --> hook-common/src/kafka_messages/app_metrics.rs:120:4
    |
120 | fn serialize_error_type<S>(error_type: ErrorType, serializer: S) -> Result<S::Ok, S::Error>
    |    ^^^^^^^^^^^^^^^^^^^^    ---------------------
    = note: this error originates in the derive macro `Serialize` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
I did a little searching but I don't know if this is possible, if you know a way I'm happy to come back and clean it up.